Makefile

```diff
@@ -0,0 +1,28 @@
+RELEASE_NAME=9.2.3.Final
+DRY_RUN=true
+PUSH_ORIGIN=origin
+
+_replace_version_in_dockerfile:
+	sed -i "s/ENV INFINISPAN_VERSION.*/ENV INFINISPAN_VERSION $(RELEASE_NAME)/g" ./server/Dockerfile
+.PHONY: _replace_version_in_dockerfile
+
+_commit_changes:
+	git commit -a -m "$(RELEASE_NAME) upgrade"
+.PHONY: _commit_changes
+
+_tag_changes:
+	git tag "$(RELEASE_NAME)"
+.PHONY: _tag_changes
+
+_push_changes:
+ifeq ($(DRY_RUN), false)
+	git push $(PUSH_ORIGIN)
+else
+	$(info DRY_RUN is set to true. Skipping...)
+endif
+.PHONY: _push_changes
+
+release: _replace_version_in_dockerfile _commit_changes _tag_changes _push_changes
```

Comment:
Here's where I was suggesting to call the tests before releasing: create a target called ```run_tests``` that executes ``` ./ci_check.sh && ci_openshift.sh```, before committing and tagging,  and abort everything if it fails.

Reply:
No problem, I can do that. 

My argument for skipping the tests is that you don't want to solve unstable CI problems when you're doing the release. So maybe let me add a switch that would disable running those tests (just in case).

Reply:
So we should make sure the tests are stable :)  
Btw, we'll need ```docker``` with insecure registry configured in the slaves in order to run those tests.

Reply:
> So we should make sure the tests are stable :)

+1 but the thing is that they have been running on Travis for quite a long time. They are stable there. But are they going to be on Central CI? I don't know. I hope so but this changes the game a bit.

> Btw, we'll need docker with insecure registry configured in the slaves in order to run those tests.

That shouldn't be necessary. On my local box (`Docker version 1.13.1, build 7f1fa5c-unsupported`), Docker emits a warning but works just fine.

Reply:
```ci_check.sh``` runs fine on any docker version, but ```ci_openshift.sh``` needs to push the image to the internal registry. Also, If I'm not wrong, openshift requires registry api v2+, which also implies docker 1.15 or superior

Reply:
The test runs still on OCP 3.6, which is both v1 and v2 IIRC ;)

Reply:
Ha! The `ci_openshift.sh` checks works fine in Jenkins. Just check the logs: https://ci.infinispan.org/job/Infinispan%20Docker%20Release/16/console

Reply:
Are you sure your local box doesn't have "insecure-registry" configured in the docker daemon? I was never able to run ```ci_openshift``` without it